x/interchainquery/keeper/msg_server.go

```diff
@@ -93,6 +94,34 @@ func (k Keeper) VerifyKeyProof(ctx sdk.Context, msg *types.MsgSubmitQueryRespons
 	return nil
 }
 
+// Checks if the query timed out, and if so, checks the timeout policy
+// Returns whether the callback should be executed (only occurs if explicitly indicated)
+func (k Keeper) CheckForQueryTimeout(ctx sdk.Context, query types.Query) (executeCallback bool, err error) {
+	if !query.HasTimedOut(ctx) {
+		return true, nil
```

Comment:
> So if the query hasn't timed out we always want to execute the callback?
correct

This is in the msg server tx so this is only called when a query is submitted. Are you saying that your nervous we might accidentally call this function elsewhere? If so, I think that's probably not a good reason to remove it - but it could be solved with a better name.

I'd also be fine with removing it and using something like the approach in the above comment since it seemed to have confused both you and RIley

Reply:
yeah I think part of the confusion was also that the execution was kinda half baked in that it only executes the query retry but did not execute the callbacks.

Sounds like we're on the same page that the refactor is better